src/pretix/control/views/global_settings.py

```diff
@@ -26,13 +35,78 @@ def form_valid(self, form):
         return super().form_valid(form)
 
     def form_invalid(self, form):
-        messages.error(self.request, _('Your changes have not been saved, see below for errors.'))
+        messages.error(
+            self.request, _('Your changes have not been saved, see below for errors.')
+        )
         return super().form_invalid(form)
 
     def get_success_url(self):
         return reverse('control:admin.global.settings')
 
 
+class SSOView(AdministratorPermissionRequiredMixin, FormView):
+    template_name = 'pretixcontrol/global_sso.html'
+    form_class = SSOConfigForm
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        oauth_applications = OAuthApplication.objects.all()
+        context['oauth_applications'] = oauth_applications
+        return context
+
+    def form_valid(self, form):
+        url = form.cleaned_data['redirect_url']
+
+        try:
+            result = self.create_oauth_application(url)
+        except (IntegrityError, ValidationError, ObjectDoesNotExist) as e:
+            error_type = type(e).__name__
+            error_message = str(e)
```

Comment:
Just pass `e` to the logger. The `%s` formatter already tell logger to "stringify" the passed arguments.
Don't wrap `str(e)` because it will make the "stringify" operation run before we need to build the log message.